src/test/login.test.ts

```diff
@@ -1,20 +1,84 @@
+  expect(createdUser.name).to.equal(user.name);
+  expect(createdUser.id).to.match(/^\d+$/);
+}
 
 describe('Login Mutation', () => {
-  it('should return mocked data in the specified format', async () => {
-    const params = { email: 'mocked@fake.com', password: 'v4l1dp4ss' };
-    const result = await loginRequest({ input: params });
+  afterEach(async () => {
+    await appDataSource.getRepository(User).clear();
+  });
+
+  it('should not complete login due to email not found', async () => {
+    const user = {
+      birthDate: '09/11/2022',
+      email: 'test@fake.com',
+      name: 'test user',
+      password: 'c0rr3ctp4ass',
+    };
+    await createUserAndAssert(user);
+
+    const loginParams = { email: 'wrong@fake.com', password: 'v4l1dp4ss' };
+
+    const response = await loginRequest({ input: loginParams });
+
+    expect(response).to.exist;
+
+    const customError: CustomError = response?.data?.errors[0];
+
+    expect(customError.message).to.equal('Invalid email.');
+    expect(customError.code).to.equal(422);
+    expect(customError.additionalInfo).to.equal('No user was found with the corresponding email.');
+  });
+
+  it('should not complete login due to incorrect password', async () => {
+    const user = {
+      birthDate: '09/11/2022',
+      email: 'test@fake.com',
+      name: 'test user',
+      password: 'c0rr3ctp4ass',
+    };
+    await createUserAndAssert(user);
+
+    const loginParams = { email: 'test@fake.com', password: '1nc0rrectp4ass' };
+
+    const response = await loginRequest({ input: loginParams });
+
+    const customError = response?.data?.errors[0];
+
+    expect(customError.message).to.equal('Invalid password.');
+    expect(customError.code).to.equal(422);
+    expect(customError.additionalInfo).to.equal('Incorrect password for the given email.');
+  });
+
+  it('should complete login', async () => {
+    const user = {
+      birthDate: '09/11/2022',
+      email: 'test@fake.com',
+      name: 'test user',
+      password: 'c0rr3ctp4ass',
+    };
+    await createUserAndAssert(user);
+
+    const loginParams = { email: 'test@fake.com', password: 'c0rr3ctp4ass' };
 
-    expect(result).to.exist;
+    const response = await loginRequest({ input: loginParams });
 
-    const unwrappedResponse: { user: User; token: string } = result?.data?.data?.login;
+    const loginResponse = response?.data?.data?.login;
 
-    expect(unwrappedResponse.token).to.be.a('string');
-    expect(unwrappedResponse.user.id).to.match(/^\d+$/);
-    expect(unwrappedResponse.user.email).to.equal(params.email);
-    expect(unwrappedResponse.user.birthDate).to.be.a('string');
-    expect(unwrappedResponse.user.name).to.be.a('string');
+    expect(loginResponse.token).to.be.a('string');
+    expect(loginResponse.user.id).to.match(/^\d+$/);
+    expect(loginResponse.user.email).to.equal(user.email);
+    expect(loginResponse.user.birthDate).to.equal(user.birthDate);
+    expect(loginResponse.user.name).to.be.equal(user.name);
```

Comment:
```suggestion
    expect(loginResponse.user).to.be.deep.eq({ id: user.id, ...})
```
agora que você tem o usuário definido pode comparar todos os campos ja direto com o deep.eq

Reply:
Como não há como deixar um valor fixo para checar o id, eu estava usando o match. Porém, ao usar o deep.eq entendo que teria que especificar o valor do campo. Haveria alguma forma de manter o match dentro do deep.eq? Não achei esse caso na doc.

Reply:
isso! A ideia seria no próprio it você ter o usuário salvo no banco para poder comparar